zuchini-examples/zuchini-examples-mockmvc/src/test/java/org/zuchini/examples/mockmvc/MockMvcSteps.java

```diff
@@ -0,0 +1,102 @@
+package org.zuchini.examples.mockmvc;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.result.HeaderResultMatchers;
+import org.springframework.test.web.servlet.result.JsonPathResultMatchers;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.zuchini.annotations.Given;
+import org.zuchini.annotations.Then;
+import org.zuchini.annotations.When;
+import org.zuchini.runner.tables.Datatable;
+import org.zuchini.spring.ScenarioScoped;
+
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.request;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@Component
+@ScenarioScoped
+public class MockMvcSteps {
+    @Autowired
+    private MockMvc mvc;
+
+    private HttpHeaders requestHeaders = new HttpHeaders();
+    private ResultActions resultActions;
+
+    private static MultiValueMap<String, String> toMultiValueMap(Datatable datatable) {
+        LinkedMultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+        for (List<String> header : datatable.getRows()) {
+            Iterator<String> iterator = header.iterator();
+            if (!iterator.hasNext()) {
+                throw new IllegalStateException("Name is required in datatable");
+            }
+
+            String headerName = iterator.next();
+            while (iterator.hasNext()) {
+                String headerValue = iterator.next();
+                if (!headerValue.isEmpty()) {
+                    map.add(headerName, headerValue);
+                }
+            }
+        }
+        return map;
+    }
+
+    @Given("^the following request headers:$")
+    public void givenRequestHeaders(Datatable datatable) {
+        requestHeaders.putAll(toMultiValueMap(datatable));
+    }
+
+    @When("^a (GET|POST|PUT|DELETE) request to \"([^\"]+)\" is executed$")
```

Comment:
HEAD, PATCH, OPTIONS, TRACE? :disappointed:

Reply:
Just an example :)

We could discuss a generic vocabulary for rest/mvc tests separately